script/test.js

```diff
@@ -0,0 +1,128 @@
+function compareArray (array, array2) {
+    for (var i = 0; i < array.length; i++) {
+        for(var j = 0; j < array.length; j++) {
+            if (array[i][j] != array2[i][j]) {
+                return false;
+            }
+        }
+    }
+    return true;
+}
+
+describe ("MAZE", function() {
+    
+    var testMaze =[
+        [0,1,0,0,0],
+        [0,1,0,1,0],
+        [0,0,0,1,0],
+        [1,1,0,1,0],
+        [1,1,1,1,0]
+        ];
+    
+    var graphVertices = [
+        [1,0,7,8,9],
+        [2,0,6,0,10],
+        [3,4,5,0,11],
+        [0,0,6,0,12],
+        [0,0,0,0,13]
+        ];
+        
+    var incorrectMaze = [
+        [0,1,0],
+        [0,1,0],
+        [0,0,1]
+        ];
+        
+    var mazeWithWay = [
+        ['GO',1,'GO','GO','GO'],
+        ['GO',1,'GO',1,'GO'],
+        ['GO','GO','GO',1,'GO'],
+        [1,1,0,1,'GO'],
+        [1,1,1,1,'GO']
+        ];
+        
+    it("Проверяет начало и конец лабиринта", function() {
+        assert.equal(crossChecking(testMaze),"correct maze")
+    });
+    
+    it ("Проверяет, что лабиринт непроходим", function() {
+        assert.equal(crossChecking(incorrectMaze), "maze is impassable")
+    });
+    
+    it("Создаёт пустой двумерный массив", function() {
+        var k = checkEmptyArray(testMaze); 
+        function checkEmptyArray(array) {
+            var newArray = createMatrix(array);
+            newArray.forEach(function(item, i, arr) {
+                if (item != 0){
+                    return false;
+                }
+            })
+            return true;
+        }
```

Comment:
hm...

you should keep your test as simple as possible
inner function is not good idea here